x/crosschain/keeper/gas_payment.go

```diff
@@ -43,42 +56,43 @@ func (k Keeper) PayGasAndUpdateCctx(
 
 // ChainGasParams returns the params to calculates the fees for gas for a chain
 // tha gas address, the gas limit, gas price and protocol flat fee are returned
-func (k Keeper) ChainGasParams(
-	ctx sdk.Context,
-	chainID int64,
-) (gasZRC20 ethcommon.Address, gasLimit, gasPrice, protocolFee math.Uint, err error) {
-	gasZRC20, err = k.fungibleKeeper.QuerySystemContractGasCoinZRC20(ctx, big.NewInt(chainID))
+func (k Keeper) ChainGasParams(ctx sdk.Context, chainID int64) (ChainGasParams, error) {
+	gasZRC20, err := k.fungibleKeeper.QuerySystemContractGasCoinZRC20(ctx, big.NewInt(chainID))
 	if err != nil {
-		return gasZRC20, gasLimit, gasPrice, protocolFee, err
+		return ChainGasParams{}, errors.Wrap(err, "unable to get system contract gas coin")
 	}
 
 	// get the gas limit
-	gasLimitQueried, err := k.fungibleKeeper.QueryGasLimit(ctx, gasZRC20)
-	if err != nil {
-		return gasZRC20, gasLimit, gasPrice, protocolFee, err
+	gasLimit, err := k.fungibleKeeper.QueryGasLimit(ctx, gasZRC20)
+	switch {
+	case err != nil:
+		return ChainGasParams{}, errors.Wrap(err, "unable to get gas limit")
+	case gasLimit == nil:
+		return ChainGasParams{}, errors.New("gas limit is nil")
```

Comment:
NIT : Can consider defining a type for the new errors

Reply:
https://github.com/zeta-chain/node/pull/2483/commits/4193e83b33a82eb2f701249f7fc1492a945e5987